### PR TITLE
Add info on installation of software with dependent modules.

### PR DESCRIPTION
**Pull request by jkwmoore**
This PR also corrects a spurious TOCtree inclusion and fixes a reference to our live clusters.

### CODE REVIEW

File: hpc/modules.rst [hunk 1]
```diff
@@ -317,6 +328,28 @@ If you no longer want to to have access to your own module files then you can ru
...
 
+-----
+
+Compiling software dependent on modules
+---------------------------------------
+
+In most cases, if you are compiling software with dependencies on modules the only actions you need to take are to load the required modules, run any ``./configure`` or **CMake** steps 
+and then run the ``make``, ``make check`` (if available) and ``make install`` commands to build, check and install the software.
+
+Once the software is installed, each time you use the software you must first load the modules used to compile it to make the required libraries and other files used during the compilation available to the program.
```
Comment (Carldkennedy) [inline hpc/modules.rst:339]:
```suggestion
Once the software is installed, each time you use the software you must first load the modules used to compile it. This is necessary to make the required libraries and other files used during the compilation available to the program.
```

File: hpc/modules.rst [hunk 2]
```diff
@@ -317,6 +328,28 @@ If you no longer want to to have access to your own module files then you can ru
...
+
+For more detailed information on the software installation process, please see: :ref:`installing-personal-software-installations`.
+
+If there are no preconfiguration steps available to generate a suitable **makefile** based on the current shell environment after loading modules or an example **makefile** for editing is provided,
```
Comment (Carldkennedy) [inline hpc/modules.rst:343]:
```suggestion
If there are no preconfiguration steps available to generate a suitable **makefile** based on the current shell environment after loading modules, or if an example **makefile** for editing is not provided, 
```

**Review by Carldkennedy — COMMENTED** [review]
LGTM with a  few small changes

File: hpc/modules.rst (hunk 2 above)
Comment (jkwmoore) [reply hpc/modules.rst:343]: I did mean if a editable file is provided here. 

One not being provided would also require them to get the same additional info mind you, but it makes  for a really long sentence. Refactor into bullets?

Comment (jkwmoore) [reply hpc/modules.rst:343]: e.g.

If:


- There are no preconfiguration steps available to generate a suitable makefile based on the current shell environment after loading modules.

- An example makefile for editing is provided.

- No makefile is provided.

You will have to construct/edit your own customised makefile which may have to reference specific libraries and paths.

Comment (Carldkennedy) [reply hpc/modules.rst]: Looks good. Maybe

You will need to construct/edit your own customised makefile which may have to reference specific libraries and paths if:
.
.
.

Comment (Carldkennedy) [reply hpc/modules.rst]: Or the way you just did it :)

**Review by Carldkennedy — APPROVED** [review]
LGTM

**Pull request merged**